### DISCUSSION TITLE
Show history of recent channels across teams

### DISCUSSION THREAD

**Discussion by andrewbrown00** [mattermost/community#69] [Enhancements]
Once a user is part of multiple teams, it becomes challenging to quickly find recently viewed channels across your teams. It would be helpful to provide users a way to view the history of recently viewed channels across teams at the global (server) level.

**Potential solutions**

1. Add an additional global level control (icon) in close proximity to the more recently surfaced history arrows in the desktop app UI. The history icon would live on its own in the browser version since we hide history arrows and leverage the browser's native ones.

**Forward and Back arrows for referecne**
![image](https://user-images.githubusercontent.com/48559658/123172435-27c83500-d44b-11eb-9928-2464cd36234c.png)

2. Include the recent channel history across teams in the 'Switch Channels' modal. 
   - _This doesn't feel like a great solution as this is more contextual to the single teams' recently viewed channels._

**Comment by enahum** [comment]
So this would switch the team as well? Isn't this doable with the history on a browser? How will this work on mobile, or not intended to be use on mobile devices (including tablets)

**Reply by Willyfrog** [reply]
I guess the idea is to be able to jump directly between the last few without clicking multiple times on the history button or trying to figure out if the title is the one you want and with a nicer screen.

I don't see this as a webapp only, in fact it would be nice to have this as a quick access in mobile so you don't have to scroll through a big list of channels.

**Reply by larkox** [reply]
Actually, I feel this even more important for mobile, since the back navigation gets you out of the app, not to the previous channel.

**Reply by Willyfrog** [reply]
yeah, after writing it, I thought this would be a nice mobile first feature to have, and let the desktop catch up later

**Reply by andrewbrown00** [reply]
@enahum correct, it would switch the team. It is doable in the history of the browser, just less discoverable. I was thinking we could introduce a control (icon button) that opens a menu. In the list, users will see 'recent' activity across teams with a visual indicator to which team it's from; creating quick recognition.

💯  will have mobile support. Need to think through how this will work on mobile and where the control will live.

**Very rough example**
![image](https://user-images.githubusercontent.com/48559658/123272871-255bee80-d4d0-11eb-9737-fc31374efcd8.png)

**Comment by chetanyakan** [comment]
This can also include unread channels accross teams

**Comment by esethna** [comment]
Related is the recent changes to the channel switcher to sort recent channels to the top. If this worked across teams that would solve this without any new UI I think. 
![image](https://user-images.githubusercontent.com/14333569/123338024-fa3bc400-d4fc-11eb-82b9-3d54ee7fd57e.png)
